src/VisualStudio/Core/Test/Snippets/SnippetTestState.vb

```diff
@@ -149,31 +163,119 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Snippets
             End Sub
         End Class
 
+        <ExportLanguageService(GetType(ISnippetExpansionClientFactory), LanguageNames.CSharp, ServiceLayer.Test)>
+        <[Shared]>
+        <PartNotDiscoverable>
+        Friend Class MockCSharpSnippetExpansionClientFactory
+            Inherits AbstractSnippetExpansionClientFactory
+
+            Private ReadOnly _threadingContext As IThreadingContext
+            Private ReadOnly _signatureHelpControllerProvider As SignatureHelpControllerProvider
+            Private ReadOnly _editorCommandHandlerServiceFactory As IEditorCommandHandlerServiceFactory
+            Private ReadOnly _editorAdaptersFactoryService As IVsEditorAdaptersFactoryService
+            Private ReadOnly _argumentProviders As ImmutableArray(Of Lazy(Of ArgumentProvider, OrderableLanguageMetadata))
+            Private ReadOnly _editorOptionsService As EditorOptionsService
+
+            <ImportingConstructor>
+            <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+            Public Sub New(
+                threadingContext As IThreadingContext,
+                signatureHelpControllerProvider As SignatureHelpControllerProvider,
+                editorCommandHandlerServiceFactory As IEditorCommandHandlerServiceFactory,
+                editorAdaptersFactoryService As IVsEditorAdaptersFactoryService,
+                <ImportMany> argumentProviders As IEnumerable(Of Lazy(Of ArgumentProvider, OrderableLanguageMetadata)),
+                editorOptionsService As EditorOptionsService)
+                MyBase.New(threadingContext)
+
+                _threadingContext = threadingContext
+                _signatureHelpControllerProvider = signatureHelpControllerProvider
+                _editorCommandHandlerServiceFactory = editorCommandHandlerServiceFactory
+                _editorAdaptersFactoryService = editorAdaptersFactoryService
+                _argumentProviders = argumentProviders.ToImmutableArray()
+                _editorOptionsService = editorOptionsService
+            End Sub
+
+            Protected Overrides Function CreateSnippetExpansionClient(textView As ITextView, subjectBuffer As ITextBuffer) As AbstractSnippetExpansionClient
+                Return New MockSnippetExpansionClient(
+                _threadingContext,
+                Guids.CSharpLanguageServiceId,
+                textView,
+                subjectBuffer,
+                _signatureHelpControllerProvider,
+                _editorCommandHandlerServiceFactory,
+                _editorAdaptersFactoryService,
+                _argumentProviders,
+                _editorOptionsService)
+            End Function
+        End Class
+
+        <ExportLanguageService(GetType(ISnippetExpansionClientFactory), LanguageNames.VisualBasic, ServiceLayer.Test)>
```

Comment:
We should be able to do a single export for both languages, right?

Reply:
➡️ Also leaving this one for https://github.com/dotnet/roslyn/pull/71752#discussion_r1465190295

Reply:
➡️ Only one ExportLanguageService attribute can be applied to a type